code/go/0chain.net/blobbercore/allocation/deletefilechange.go

```diff
@@ -76,7 +81,7 @@ func (nf *DeleteFileChange) CommitToFileStore(ctx context.Context) error {
 					Where("allocation_id=? AND content_hash=?", nf.AllocationID, res.ContentHash).
 					Count(&count)
 
-				if count != 0 {
+				if count != 0 && res.ThumbnailHash == "" {
```

Comment:
I would suggest to use `res.ThumbnailSize ==0` instead of `res.ThumbnailHash == ""`